src/web/src/controllers/Schedule.js

```diff
@@ -245,6 +253,10 @@ class Schedule {
 
       this.selectedSections.push(courseSection);
       console.log(`Added new courseSection ${JSON.stringify(courseSection)}`);
+      if (this.totalCredits()+course.max_credits>21){
+        console.error("Credit Cap");
+        throw {type:"Credit Conflict"};
```

Comment:
Lets not throw an error here since this behavior is expected.

Reply:
Then what do you suggest on how to handle this?

Reply:
We can remove these lines

Reply:
My problem is then how should I determine when to call the function I implemented in NotificationMixing.js? I call the function in NotificationMixing.js when I catch the error with type "credit conflict" when the function addCourseSection is called.

Reply:
you should be able to import NotificationMixin and directly call `NotificationMixin.notifyWarningCreditOverflow`

`notifyCreditConflict -> notifyWarningCreditOverflow`

Reply:
Unfortunately that won't work because the `NotificationMixin` needs to be applied to a `Vue` instance to function. This was my implementation mistake. Bubbling up an error to trigger the notification should work here, you'll just have to make sure that the course saving logic still runs.